src/test/java/org/cloudfoundry/promregator/SpringBootLoadPropertiesForTesting.java

```diff
@@ -1,56 +1,50 @@
 package org.cloudfoundry.promregator;
 
 
-import org.junit.AfterClass;
-import org.junit.Assert;
-import org.junit.ClassRule;
-import org.junit.Test;
-import org.junit.contrib.java.lang.system.EnvironmentVariables;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
 
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.not;
 
 @SpringBootTest(classes = { SpringBootLoadPropertiesForTestingSpringApplication.class })
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
 @ContextConfiguration(initializers = { ConfigFileApplicationContextInitializer.class })
 @ActiveProfiles(profiles= {"springBootLoadPropertiesForTesting"})
+@TestPropertySource(properties = "encrypt.key=mySecretKey")
```

Comment:
Eliminated the `EnvironmentVariables` approach, to just using Spring properties directly.

Reply:
Thanks for the explanation. I am not 100% convinced yet that this will work out properly. I will check further on this to see if I am right or wrong and will let you know the result.

Reply:
Luckily that works correctly.